### PR TITLE
Addition of amr modules

### PR DESCRIPTION
**Pull request by ryanjameskennedy**
# Description
_Summary of the changes made:_
* Added amrfinder
* Change "phenotype" references to "element type" as it's more accurate
* Create prp functions to parse amrfinder output
* Add amrfinder output columns to `GeneBase` pydantic class
* Fix `publishDir` of each module
* Update deploy files
* Fix module output names to be `${sampleName}_${software}.${suffix}`
* Remove ariba references

## Primary function of PR
- [ ] Hotfix
- [ ] Minor functionality improvement
- [x] Major functionality improvement / New type of analysis
- [ ] Backward-breaking functionality

# Testing
prp was tested on the ouput generated from amrfinder's output. The entire pipeline was thereafter run using illumina PE saureus data and worked as expected.

### CODE REVIEW

**Comment by ryanjameskennedy** [comment]
In the process of updating prp to be able to parse the amrfinderplus output

File: deploy/deploy_references.sh [hunk 1]
```diff
@@ -88,3 +85,5 @@ if [[ -f $ref && -f $refamb && -f $refann && -f $refbwt && -f $refpac && -f $ref
...
 if [[ -f $mlst && -f $mlstndb && -f $mlstnhd && -f $mlstnhi && -f $mlstnhr && -f $mlstnin && -f $mlstnog && -f $mlstnos && -f $mlstnot && -f $mlstnsq && -f $mlstntf && -f $mlstnto ]]; then echo "BLAST indexes exists!"; else echo "ERROR: BLAST indexes do not exist!!! Please report this to JASEN issues."; fi
+
+AMR.LIB,AMRProt,AMRProt-mutation.tab,AMRProt-suppress,AMRProt-susceptible.tab,AMR_CDS,AMR_DNA-Campylobacter,AMR_DNA-Campylobacter.tab,AMR_DNA-Clostridioides_difficile,AMR_DNA-Clostridioides_difficile.tab,AMR_DNA-Enterococcus_faecalis,AMR_DNA-Enterococcus_faecalis.tab,AMR_DNA-Enterococcus_faecium,AMR_DNA-Enterococcus_faecium.tab,AMR_DNA-Escherichia,AMR_DNA-Escherichia.tab,AMR_DNA-Klebsiella_oxytoca,AMR_DNA-Klebsiella_oxytoca.tab,AMR_DNA-Neisseria_gonorrhoeae,AMR_DNA-Neisseria_gonorrhoeae.tab,AMR_DNA-Salmonella,AMR_DNA-Salmonella.tab,AMR_DNA-Staphylococcus_aureus,AMR_DNA-Staphylococcus_aureus.tab,AMR_DNA-Streptococcus_pneumoniae,AMR_DNA-Streptococcus_pneumoniae.tab,ReferenceGeneCatalog.txt,ReferenceGeneHierarchy.txt,amr_targets.fa,changelog.txt,changes.txt,database_format_version.txt,fam.tab,taxgroup.tab,version.txt
```
Comment (sylvinite) [inline deploy/deploy_references.sh:89]: Accidental copypaste?

File: configs/nextflow.hopper.config [hunk 2]
```diff
@@ -8,16 +8,14 @@ singularity{
 params {
 	root = "/fs1/pipelines/JASEN"
 	// DATABASES //
+	amrfinderDb = "${root}/assets/amrfinder_db/2023-04-17.1"
```
Comment (sylvinite) [inline configs/nextflow.hopper.config:11]: Could this be renamed to a generic name so the config doesn't have to be updated for every version

File: configs/nextflow.base.config [hunk 3]
```diff
@@ -1,14 +1,12 @@
 params {
 	root = "/filepath/to/JASEN" //edit
+	amrfinderDb = "${root}/assets/amrfinder_db/2023-04-17.1"
```
Comment (sylvinite) [inline configs/nextflow.base.config:3]: Same as the note on the hopper config

File: container/build_container.sh [hunk 4]
```diff
@@ -7,7 +7,8 @@ declare -A containers=(
     [kraken2]=https://depot.galaxyproject.org/singularity/kraken2:2.1.2--pl5321h9f5acd7_3
     [bracken]=https://depot.galaxyproject.org/singularity/bracken:2.8--py39hc16433a_0
     [samtools]=https://depot.galaxyproject.org/singularity/samtools:1.17--h00cdaf9_0
-    [ariba]=https://depot.galaxyproject.org/singularity/ariba:2.14.6--py39h8f06ca0_2
```
Comment (sylvinite) [inline container/build_container.sh:10]: There's a bunch of places that still mention ariba. Quick search showed the readme, stale references in config, deploy_reqs.sh. Probably want to either delete them or chuck them in a legacy/ repo.

**Review by sylvinite — COMMENTED** [review]
Wow! Nice!

Added some notes on small things. Everything else looks stellar

File: deploy/deploy_references.sh (hunk 1 above)
Comment (ryanjameskennedy) [reply deploy/deploy_references.sh:89]: Indeed!

File: configs/nextflow.hopper.config (hunk 2 above)
Comment (ryanjameskennedy) [reply configs/nextflow.hopper.config:11]: `amrfinderDb = "${root}/assets/amrfinder_db/latest"`?

File: container/build_container.sh (hunk 4 above)
Comment (ryanjameskennedy) [reply container/build_container.sh:10]: Will do!

**Comment by ryanjameskennedy** [comment]
Still getting the following `pydantic` error regarding `prp`:
```
Input failed Validation
1 validation error for PipelineResult
phenotypeResult
  field required (type=value_error.missing)
Aborted
```

I know how to fix it, it will just take a bit more time. Almost there!

**Comment by ryanjameskennedy** [comment]
The resfinder output isn't working but amrfinder is working!

**Pull request merged**